src/ar_storage.erl

```diff
@@ -385,45 +391,22 @@ lookup_tx_filename(ID) ->
 enough_space(Bytes) ->
 	(ar_meta_db:get(disk_space)) >= (Bytes + ar_meta_db:get(used_space)).
 
-%% @doc Sum up the sizes of the files written by the Arweave client.
-calculate_used_space() ->
-	DataDir = ar_meta_db:get(data_dir),
-	{ok, CWD} = file:get_cwd(),
-	Dirs = [
-		filename:join(DataDir, ?BLOCK_DIR),
-		filename:join(DataDir, ?ENCRYPTED_BLOCK_DIR),
-		filename:join(DataDir, ?HASH_LIST_DIR),
-		filename:join(DataDir, ?TX_DIR),
-		filename:join(DataDir, ?TX_INDEX_DIR),
-		filename:join(DataDir, ?WALLET_DIR),
-		filename:join(DataDir, ?WALLET_LIST_DIR),
-		filename:join(CWD, ?LOG_DIR)
-	],
-	lists:foldl(
-		fun(Dir, Acc) ->
-			Acc + calculate_used_space(Dir)
-		end,
-		0,
-		Dirs
-	).
+%% @doc Calculate the available space in bytes on the data directory disk.
+calculate_disk_space() ->
+	{_, Size, _} = get_data_dir_disk_data(),
+	Size * 1024.
 
-%% @doc Sum up the sizes of the files located directly (not in subdir) under the directory.
-calculate_used_space(Directory) ->
-	filelib:fold_files(
-		Directory,
-		"",
-		false,
-		fun(File, Acc) -> Acc + filelib:file_size(File) end,
-		0
-	).
+%% @doc Calculate the used space in bytes on the data directory disk.
+calculate_used_space() ->
+	{_, Size, UsedPercentage} = get_data_dir_disk_data(),
+	math:ceil(Size * UsedPercentage / 100 * 1024).
```

Comment:
Why are we multiplying by 1024 here?

Reply:
`disksup` returns KBytes. Would you prefer to have this conversion inside `get_data_dir_disk_data`?

Reply:
How about just renaming the variable to KByteSize then?